Documentation/Contributing.md

```diff
@@ -134,66 +93,87 @@ There are three ways of creating a template.
-        * description: A description of this category.
-        * order: The sort order of category.
-3. Now we need to modify template settings to associate it together.
-4. Go to your template and open settings.json file
-    ```jsonc
-    {
-        "$schema": "..//schema/settings.json",
-        "name": "Bracket Retention",
-        "author": "Microsoft",
-        "galleries": [
-            {
-                "type": "workbook",
-                "resourceType": "microsoft.insights/components",
-                "order": 400
-            },
-            // This is the new section to add:
-            {
-                "type": "workbook",
-                "resourceType": "Azure Monitor",
-                "categoryKey": "YourSampleUniqueCategoryKey",
-                "order": 400
-            }
-        ]
-    }
-    ```
-    **Note that the second item in the galleries array, it has a "categoryKey". It should be match with a "key" in a virtual category.**
+            ]
+        }
+    ]
+}
+```
+
+* `$schema`: The link to the gallery schema. This should be `"https://raw.githubusercontent.com/microsoft/Application-Insights-Workbooks/master/schema/gallery.json"`
+
+* `version`: Gallery version (eg. `TemplateGallery/1.0`)
+
+* `categories`: A list of categories for this gallery. *Note*: The order of categories in this list determines the order of the categories that will appear in this gallery
+
+    * `id`: The ID for the category. This field will not be localized
+
+    * `name`: The name of the category. This field will be localized
+    
+    * `templates`: A list of templates for the category. *Note*: The order of the templates in this list determines the order of the templates that will appear in this category
+
+        * `id`: The ID for the template. This ID should be the path to your template folder (eg. `Workbooks/Performance/Apdex`)
+
+        * `name`: The name of the template. This field will be localized
+
+        * `description`: The description for the template. This field will be localized
+
+        * `author`: Author for this template (eg. `Microsoft`)
+
+        * `icon`: Optional. If you don't specify "icon" property, it will use the default icon. Otherwise, specify the name of icon file that is located under the template folder.
+
+        * `tags`: Optional. You can specify a list of tags that describes the template.
+
+        * `isPreview`: Optional. Flag to mark the template as preview. See [Testing Preview Workbook Templates](#testing-preview-workbook-templates) for more details
+
+For more details on the schema of the gallery file, view the [Gallery JSON schema](./schema/gallery.json).
+
+## How to create and name a gallery file
+Create a gallery file with the above schema under the `\gallery` folder.
+ 
+ The gallery file name must follow the following convention:
+
+ `{type}`-`{resourceType}`.json
+
+ `{type}` is the Workbook type like 'tsg', 'performance', 'insights', 'workbook', etc.
```

Comment:
probably add a comment that these are known types and controlled by the workbooks team, they aren't arbitrary?  we should probably add a validation, because i think right now the workload insights has some "galleries" that aren't actually real, there's no workbook type "workloads-onboarding" or "workloads-shared" workbook types.

though given the type can contain `-` i guess it might be hard to extract out the names?

i guess part of the problem with any naming here is that parts would have / in them (like resource types), and so all the separators are just -? not sure i have a better way to do it without adding more special characters to file names.  unless we made a folder per workbook type or something?